sway-lib-std/src/assert.sw

```diff
@@ -10,3 +11,17 @@ pub fn assert(a: bool) {
         ()
     }
 }
+
+/// A wrapper for `assert` that allows logging a custom value `v` if condition `c` is not true.
+/// This will then revert with the value `42`, which indicates that you should look at the previous logd receipt for further debugging clues. Note that the SDK will not currently decode this logd receipt for you.
+pub fn require<T>(c: bool, v: T) {
+    if !c {
+        let size = size_of::<T>();
```

Comment:
As I've simply used `logd` for all types, I don't think there's a need to call `is_reference_type()` if we just need the size to pass to the `logd` op.

Reply:
Note: we can replace the logging here with the generic `log` func in the stdlib (coming soon).

Reply:
`LOGD` expects `r1` below to be a pointer so if it isn't a reference type it'll try to read memory from somewhere weird and not make sense.  The VM spec doesn't say what happens if you try to log invalid memory but it might panic?

So I think `LOG` for non-ref and `LOGD` for ref really should be done here, unless `log()` is very imminent and will Do The Right Thing?

Reply:
Makes sense. I'll fix this to do the right thing, and swap it out to use the generic `log()` when it lands.
Otherwise this is one more PR on hold. This way at least `require` will be usable.